lib/js/jsbigints.nim

```diff
@@ -0,0 +1,127 @@
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func `$`*(this: JsBigInt): cstring {.importjs: "#.toString()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func asIntN*(width: int; bigInteger: JsBigInt): int {.importjs: "BigInt.$1(#, #)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asIntN
+
+func asUintN*(width: int; bigInteger: JsBigInt): int {.importjs: "BigInt.$1(#, #)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asUintN
+
+func `+`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `-`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `*`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `div`*(x, y: JsBigInt): JsBigInt {.importjs: "(# / #)".}
+   ## Same as `div` but for `JsBigInt`.
+
+func `mod`*(x, y: JsBigInt): JsBigInt {.importjs: "(# % #)".}
+  ## Same as `mod` but for `JsBigInt` (uses JavaScript `BigInt() % BigInt()`).
+
+func `+=`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)", discardable.}
+
+func `-=`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)", discardable.}
+
+func `*=`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)", discardable.}
+
+func `/=`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)", discardable.}
+
+func `mod=`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)", discardable.}
+
+func `inc`*(x: JsBigInt): JsBigInt {.importjs: "(++#)", discardable.}
+  ## Same as `inc` but for `JsBigInt` (uses JavaScript `++BigInt()`).
+
+func `dec`*(x: JsBigInt): JsBigInt {.importjs: "(--#)", discardable.}
+  ## Same as `dec` but for `JsBigInt` (uses JavaScript `--BigInt()`).
+
+func `<`*(x, y: JsBigInt): bool {.importjs: "(# $1 #)".}
+
+func `<=`*(x, y: JsBigInt): bool {.importjs: "(# $1 #)".}
+
+func `==`*(x, y: JsBigInt): bool {.importjs: "(# === #)".}
+
+func `**`*(x, y: JsBigInt): JsBigInt {.importjs: "((#) $1 #)".}
+
+func `and`*(x, y: JsBigInt): JsBigInt {.importjs: "(# && #)".}
+
+func `or`*(x, y: JsBigInt): JsBigInt {.importjs: "(# || #)".}
+
+func `not`*(x: JsBigInt): JsBigInt {.importjs: "(!#)".}
+
+func `in`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `-`*(a: JsBigInt): JsBigInt {.importjs: "($1#)".}
+
+func `xor`*(x, y: JsBigInt): JsBigInt {.importjs: "(# ^ #)".}
+
+func `shl`*(a, b: JsBigInt): JsBigInt {.importjs: "(# << #)".}
+
+func `shr`*(a, b: JsBigInt): JsBigInt {.importjs: "(# >> #)".}
+
+func inc*(a, b: JsBigInt): JsBigInt {.importjs: "(# += #)", discardable.}
+
+func dec*(a, b: JsBigInt): JsBigInt {.importjs: "(# -= #)", discardable.}
+
+func `+`*(a: JsBigInt): JsBigInt {.error.} # Can not be used by design.
+  ## https://github.com/tc39/proposal-bigint/blob/master/ADVANCED.md#dont-break-asmjs
+
+
+runnableExamples:
+  let big1: JsBigInt = newBigInt(2147483647)
+  let big2: JsBigInt = newBigInt("666".cstring)
```

Comment:
this is very unfriendly, how about adding this:
```nim
func b*(a: cstring): JsBigInt {.importjs: "BigInt(#)".}

# in runnableExamples (pick a better example though)
doAssert b"-123" * b"456" == b"-56088"
```

Reply:
`big"999"` ?.